molecule.py

```diff
@@ -3,21 +3,37 @@
 from qiskit.circuit.library import excitation_preserving
 from qiskit.primitives import StatevectorEstimator
 from qiskit.quantum_info import SparseObservable, SparsePauliOp
+from qiskit_nature.second_q.formats import fcidump_to_problem
+from qiskit_nature.second_q.formats.fcidump import FCIDump
+from qiskit_nature.second_q.mappers import JordanWignerMapper
+from qiskit_nature.second_q.operators import FermionicOp
 
 import cextension
 
 if __name__ == "__main__":
-    obs = cextension.qubit_observable()
+    filename = "lih.fcidump"
+    obs = cextension.molecular_hamiltonian(filename)
     print("-- SparseObservable")
     print(obs)
 
+    # primitives V2 don't support SparseObservable yet...
+    # ... and we want to compare against Qiskit Nature's output
+    spo = SparsePauliOp.from_sparse_observable(obs)
+
+    fcidump = FCIDump.from_file(filename)
+    problem = fcidump_to_problem(fcidump)
+    mapper = JordanWignerMapper()
+    hamil = problem.hamiltonian.second_q_op()
+    hamil += sum(problem.hamiltonian.constants.values()) * FermionicOp.one()
+    op = mapper.map(hamil)
+
+    assert spo.equiv(op)
```

Comment:
We can remove this again later, but I would like to try and complete unrestricted-spin support before doing so.